packs/kubernetes/sensors/thirdpartyresource.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+import ast
+# from git@github.com:mward29/python-k8sclient.git
+
+from st2reactor.sensor.base import Sensor
+
+
+class ThirdPartyResource(Sensor):
+    def __init__(self, sensor_service, config=None):
+        super(ThirdPartyResource, self).__init__(sensor_service=sensor_service,
+                                                 config=config)
+#        self._labels = self._config['labels'].get('thirdpartyresource', [])\
+
+    def setup(self):
+        self._logger = self._sensor_service.get_logger(name=self.__class__.__name__)
+        self._logger.debug('Connecting to Kubernetes via api_client')
+        extension = self._config['extension_url']
+        KUBERNETES_API_URL = self._config['kubernetes_api_url'] + extension
+        user = self._config['user']
+        password = self._config['password']
+        verify = self._config['verify']
+        self.client = requests.get(KUBERNETES_API_URL, auth=HTTPBasicAuth(user, password),
+                                   verify=verify, stream=True)
+        pass
+
+    def run(self):
+        self._logger = self._sensor_service.get_logger(name=self.__class__.__name__)
+        self._logger.debug('Watch Kubernetes for thirdpartyresource information')
+        r = self.client
+        lines = r.iter_lines()
+        # Save the first line for later or just skip it
+#        first_line = next(lines)
+        for line in lines:
+            io = json.dumps(line)
```

Comment:
The string coming from Kubernetes can not be decoded (json.load) to a json object in its raw form. The json.dump encodes the string to a proper format so json.jump can decode it. I know it looks ackward but it works. It will not work without encoding it first.